source/common/http/codec_client.h

```diff
@@ -180,8 +182,10 @@ class CodecClient : Logger::Loggable<Logger::Id::client>,
 
   // Network::ConnectionCallbacks
   void onEvent(uint32_t events) override;
-  void onAboveWriteBufferHighWatermark() override {}
-  void onBelowWriteBufferLowWatermark() override {}
+  // Pass watermark events from the connection on to the codec which will pass it to the underlying
+  // streams.
+  void onAboveWriteBufferHighWatermark() override { codec_->onAboveWriteBufferHighWatermark(); }
```

Comment:
The connection manager has to handle callbacks from multiple parties since each filter can go over watermark, so I figured it was better to handle all the deduping in one place rather than do it both here and over there.  That's all managed by the overrun_buffers() counters.